python/ray/air/integrations/mlflow.py

```diff
@@ -1,15 +1,179 @@
 import logging
-from typing import Dict, Optional
+from types import ModuleType
+from typing import Dict, Optional, Union
 
 import ray
+from ray.air import session
+
 from ray.air._internal.mlflow import _MLflowLoggerUtil
 from ray.tune.logger import LoggerCallback
 from ray.tune.result import TIMESTEPS_TOTAL, TRAINING_ITERATION
 from ray.tune.experiment import Trial
+from ray.util.annotations import PublicAPI
+
+try:
+    import mlflow
+except ImportError:
+    mlflow = None
+
 
 logger = logging.getLogger(__name__)
 
 
+class _NoopModule:
+    def __getattr__(self, item):
+        return _NoopModule()
+
+    def __call__(self, *args, **kwargs):
+        return None
+
+
+@PublicAPI(stability="alpha")
+def setup_mlflow(
+    config: Optional[Dict] = None, rank_zero_only: bool = True, **kwargs
+) -> Union[ModuleType, _NoopModule]:
+    """Set up a MLflow session.
+
+    This function can be used to initialize an MLflow session in a
+    (distributed) training or tuning run.
+
+    By default, the MLflow experiment ID is the Ray trial ID and the
+    MLlflow experiment name is the Ray trial name. These settings can be overwritten by
+    entries in the ``mlflow`` key of the ``config`` dict.
+    Any keyword arguments will be merged into this configuration.
+
+    In distributed training with Ray Train, only the zero-rank worker will initialize
+    mlflow. All other workers will return a noop client, so that logging is not
+    duplicated in a distributed run. This can be disabled by passing
+    ``rank_zero_only=False``, which will then initialize mlflow in every training
+    worker.
+
+    This function will return the ``mlflow`` module or a noop module for
+    non-rank zero workers ``if rank_zero_only=True``. By using
+    ``mlflow = setup_mlflow(config)`` you can ensure that only the rank zero worker
+    calls the mlflow API.
```

Comment:
Should we keep the same docstring information as `mlflow_mixin`? Specifically the autologging snippet and the argument descriptions.

Reply:
Added the example. The argument descriptions were taken from the MlflowUtil, I've updated them to use the ones from the mixin instead.